app/views/ThreadMessagesView/hooks/useThreadFilter.tsx

```diff
@@ -0,0 +1,76 @@
+import { useCallback } from 'react';
+import I18n from 'i18n-js';
+
+import { Filter } from '../filters';
+import UserPreferences from '../../../lib/methods/userPreferences';
+import { showActionSheetRef } from '../../../containers/ActionSheet';
+import { CustomIcon } from '../../../containers/CustomIcon';
+import { IUser, TSubscriptionModel, TThreadModel } from '../../../definitions';
+import getFilteredThreads from '../utils/getFilteredThreads';
+
+const THREADS_FILTER = 'threadsFilter';
+
+interface IUseThreadFilter {
+	user: IUser;
+	messages: TThreadModel[];
+	subscription: TSubscriptionModel;
+	currentFilter: Filter;
+	setCurrentFilter: (filter: Filter) => void;
+	setDisplayingThreads: (threads: TThreadModel[]) => void;
+}
+
+const useThreadFilter = ({
+	user,
+	messages,
+	subscription,
+	currentFilter,
+	setCurrentFilter,
+	setDisplayingThreads
+}: IUseThreadFilter) => {
+	const initFilter = () => {
+		const savedFilter = UserPreferences.getString(THREADS_FILTER);
+		if (savedFilter) {
+			setCurrentFilter(savedFilter as Filter);
+		}
+	};
+
+	const onFilterSelected = useCallback(
+		(filter: Filter) => {
+			const displayingThreads = getFilteredThreads(user, messages, subscription, filter);
+			setCurrentFilter(filter);
+			setDisplayingThreads(displayingThreads);
+			UserPreferences.setString(THREADS_FILTER, filter);
+		},
+		[messages, subscription]
```

Comment:
Subscription is a mutating data from WatermelonDB.
Are you sure it's going to work inside this useCallback?
Did you test it?